uint8[] image_buffer
---
string caption_text